monkey/monkey_island/cc/services/telemetry/processing/post_breach.py

```diff
@@ -12,16 +16,32 @@ def process_communicate_as_new_user_telemetry(telemetry_json):
     test_new_user_communication(current_monkey, success, message)
 
 
+def process_shell_startup_file_modification_telemetry(telemetry_json):
+    modified_data = []
+    for result in telemetry_json['data']['result']:
+        temp = copy.deepcopy(telemetry_json['data'])
+        temp['result'] = result
+        modified_data.append(temp)
+    telemetry_json['data'] = modified_data
+
+
 POST_BREACH_TELEMETRY_PROCESSING_FUNCS = {
     POST_BREACH_COMMUNICATE_AS_NEW_USER: process_communicate_as_new_user_telemetry,
+    POST_BREACH_SHELL_STARTUP_FILE_MODIFICATION: process_shell_startup_file_modification_telemetry,
 }
 
 
 def process_post_breach_telemetry(telemetry_json):
-    mongo.db.monkey.update(
-        {'guid': telemetry_json['monkey_guid']},
-        {'$push': {'pba_results': telemetry_json['data']}})
-
     post_breach_action_name = telemetry_json["data"]["name"]
     if post_breach_action_name in POST_BREACH_TELEMETRY_PROCESSING_FUNCS:
         POST_BREACH_TELEMETRY_PROCESSING_FUNCS[post_breach_action_name](telemetry_json)
+
+    if type(telemetry_json['data']) is list:
+        for pba_data in telemetry_json['data']:
+            mongo.db.monkey.update(
+                {'guid': telemetry_json['monkey_guid']},
+                {'$push': {'pba_results': pba_data}})
+    else:
+        mongo.db.monkey.update(
+            {'guid': telemetry_json['monkey_guid']},
+            {'$push': {'pba_results': telemetry_json['data']}})
```

Comment:
What would happen if we left it as it was and saved a list? When implementing PBA you need to implement reporting also, so you would know how if PBA results is a list or an object.

Reply:
That would require me to make modifications on the front-end (like I'd done before here: https://github.com/guardicore/monkey/pull/731#discussion_r457879766) so that the PBA table in the Security report is generated properly. If I leave it as a list with no modifications on the front-end, it'll only show the first entry of the list.

Reply:
Why can't you parse list into a front-end readable format on `monkey/monkey_island/cc/services/attack/technique_reports/T1504.py` ? First off you can `unwind` the list into separate documents with a mongodb query. Then you can either return all documents, or additionally only match the ones you need. Either way, we're already parsing all PBA results in `monkey/monkey_island/cc/services/attack/technique_reports/`, so lets fully parse it there, instead of moving the logic into higher abstraction level classes. If you need to additionally parse some data in UI, you should do it in `T1504.js` or `T1156.js`.

Reply:
I'm making these changes here to also take into account the PBA section in the Security report tab. Sending it as a list and doing what you're suggesting will mess up that part unless I also parse it in the UI in PostBreach.js like before.